python-checks/src/main/java/org/sonar/python/checks/hotspots/PubliclyWritableDirectoriesCheck.java

```diff
@@ -0,0 +1,118 @@
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.python.PythonSubscriptionCheck;
+import org.sonar.python.api.tree.PyArgumentTree;
+import org.sonar.python.api.tree.PyCallExpressionTree;
+import org.sonar.python.api.tree.PyExpressionTree;
+import org.sonar.python.api.tree.PyNameTree;
+import org.sonar.python.api.tree.PyQualifiedExpressionTree;
+import org.sonar.python.api.tree.PyStringLiteralTree;
+import org.sonar.python.api.tree.PySubscriptionExpressionTree;
+import org.sonar.python.api.tree.Tree;
+import org.sonar.python.api.tree.Tree.Kind;
+import org.sonar.python.semantic.Symbol;
+import org.sonar.python.semantic.SymbolTable;
+
+@Rule(key = "S5443")
+public class PubliclyWritableDirectoriesCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Make sure publicly writable directories are used safely here.";
+  private static final List<String> UNIX_WRITABLE_DIRECTORIES = Arrays.asList(
+    "/tmp/", "/var/tmp/", "/usr/tmp/", "/dev/shm/", "/dev/mqueue/", "/run/lock/", "/var/run/lock/",
+    "/library/caches/", "/users/shared/", "/private/tmp/", "/private/var/tmp/");
+  private static final List<String> NONCOMPLIANT_ENVIRON_VARIABLES = Arrays.asList("tmpdir", "tmp");
+
+  private static final Pattern WINDOWS_WRITABLE_DIRECTORIES = Pattern.compile("[^\\\\]*\\\\(Windows\\\\Temp|Temp|TMP)(\\\\.*|$)", Pattern.CASE_INSENSITIVE);
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Kind.STRING_LITERAL, ctx -> {
+      PyStringLiteralTree tree = (PyStringLiteralTree) ctx.syntaxNode();
+      String stringLiteral = tree.trimmedQuotesValue().toLowerCase();
+      if (UNIX_WRITABLE_DIRECTORIES.stream().anyMatch(dir -> containsDirectory(stringLiteral, dir)) ||
+        WINDOWS_WRITABLE_DIRECTORIES.matcher(stringLiteral).matches()) {
+        ctx.addIssue(tree, MESSAGE);
+      }
+    });
+
+    context.registerSyntaxNodeConsumer(Kind.CALL_EXPR, ctx -> {
+      PyCallExpressionTree tree = (PyCallExpressionTree) ctx.syntaxNode();
+      if (isOsEnvironGetter(tree.callee(), ctx.symbolTable()) &&
+        tree.arguments().stream().anyMatch(this::isNonCompliantOsEnvironArgument)) {
+        ctx.addIssue(tree, MESSAGE);
+      }
+    });
+
+    context.registerSyntaxNodeConsumer(Kind.SUBSCRIPTION, ctx -> {
+      PySubscriptionExpressionTree tree = (PySubscriptionExpressionTree) ctx.syntaxNode();
+      if (isOsEnvironQualifiedExpression(tree.object(), ctx.symbolTable()) && tree.subscripts().expressions().stream().anyMatch(this::isNonCompliantOsEnvironArgument)) {
+        ctx.addIssue(tree, MESSAGE);
+      }
+    });
+
+  }
+
+  private static boolean containsDirectory(String stringLiteral, String dir) {
+    return stringLiteral.startsWith(dir) || stringLiteral.equals(dir.substring(0, dir.length() - 1));
+  }
+
+  private boolean isNonCompliantOsEnvironArgument(Tree tree) {
+    Tree argumentOrExpression = tree;
+    if (tree.is(Kind.ARGUMENT)) {
```

Comment:
I think we could remove this `if` and add `.map(PyArgumentTree::expression)` at line 64.